contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -1963,6 +1979,49 @@ TdsRecvTypeDatetime2(const char *message, const ParameterToken token)
 	return result;
 }
 
+/* -------------------------------
+ * TdsRecvTypeGeometry - converts external binary format to
+ * Geometry data type
+ * --------------------------------
+ */
+Datum
+TdsRecvTypeGeometry(const char *message, const ParameterToken token)
+{
+	Datum result = 0; 
+
+	//Decode binary and convert if needed 
+	StringInfo	buf = TdsGetStringInfoBufferFromToken(message, token);
+
+	//Return in Datum val
+
+	ereport(ERROR,
+							(errcode(ERRCODE_DATETIME_VALUE_OUT_OF_RANGE),
+							 errmsg("tdstypeio.c: TdsRecvTypeGeometry")));    
```

Comment:
This seems to be a debug message, let's have a better error message. Which includes "Feature not supported ....."

Reply:
Done